File: code-samples/calendar/ruby/create-calendar/create_calendar.rb
# Creating a Calendar using Nylas v3 Ruby SDK

# Importing dependencies
require 'dotenv/load'
require 'nylas'

# Initializing Nylas API Client
nylas = Nylas::Client.new(
    api_key: ENV["API_KEY"]
)

begin
    new_calendar = nylas.calendars.create(
        identifier: ENV["GRANT_ID"],
        request_body: {
            name: "My New Calendar 1",
            description: "Calendar test 1",
        }
    )
    
    # print the calendar
    puts new_calendar

rescue => error
    puts "Nylas Error: #{error}"
end

File: code-samples/calendar/java/create-calendar/CreateCalendar.kt
package com.nylas

import com.nylas.models.CreateCalendarRequest
import com.nylas.models.Calendar
import com.nylas.models.NylasApiError
import com.nylas.models.NylasSdkTimeoutError
import com.nylas.models.Response

fun main() {
    try {
        val nylas = NylasClient.Builder("API_KEY").build()

        val createCalendarRequest = CreateCalendarRequest.Builder("My Calendar") // Calendar name is required and other attributes are optional.
            .description("My calendar description") 
            .location("My calendar location") 
            .timezone("America/New_York")
            .build()

        val response: Response<Calendar> = nylas.calendars().create("GRANT_ID", createCalendarRequest)

        val calendar: Calendar = response.data

        println(
            "Created calendar with id ${calendar.id}, name ${calendar.name}, and description ${calendar.description}"
        )
    } catch (e: NylasSdkTimeoutError) {
        e.printStackTrace()
    } catch (e: NylasApiError) {
        e.printStackTrace()
    }
}
